exchanges/bybit/bybit_test.go

```diff
@@ -3683,3 +3687,58 @@ func TestGetCurrencyTradeURL(t *testing.T) {
 		assert.NotEmpty(t, resp)
 	}
 }
+
+// TestGenerateSubscriptions exercises generateSubscriptions
+func TestGenerateSubscriptions(t *testing.T) {
+	t.Parallel()
+
+	b := new(Bybit)
+	require.NoError(t, testexch.Setup(b), "Test instance Setup must not error")
+
+	b.Websocket.SetCanUseAuthenticatedEndpoints(true)
+	subs, err := b.generateSubscriptions()
+	require.NoError(t, err, "generateSubscriptions must not error")
+	exp := subscription.List{}
+	for _, s := range b.Features.Subscriptions {
+		for _, a := range b.GetAssetTypes(true) {
+			if s.Asset != asset.All && s.Asset != a {
+				continue
+			}
+			pairs, err := b.GetEnabledPairs(a)
+			require.NoErrorf(t, err, "GetEnabledPairs %s must not error", a)
+			pairs = common.SortStrings(pairs).Format(currency.PairFormat{Uppercase: true, Delimiter: ""})
+			s := s.Clone() //nolint:govet // Intentional lexical scope shadow
+			s.Asset = a
+			if isSymbolChannel(channelName(s)) {
+				for i, p := range pairs {
+					s := s.Clone() //nolint:govet // Intentional lexical scope shadow
+					switch s.Channel {
+					case subscription.OrderbookChannel:
+						s.QualifiedChannel = fmt.Sprintf("%s.%d.%s", channelName(s), s.Levels, p)
+					default:
+						s.QualifiedChannel = channelName(s) + "." + p.String()
+					}
+					s.Pairs = pairs[i : i+1]
+					exp = append(exp, s)
+				}
+			} else {
+				s.Pairs = pairs
+				s.QualifiedChannel = channelName(s)
+				exp = append(exp, s)
+			}
+		}
+	}
+	testsubs.EqualLists(t, exp, subs)
+}
+
+func TestSubscribe(t *testing.T) {
+	t.Parallel()
+	b := new(Bybit)
+	require.NoError(t, testexch.Setup(b), "Test instance Setup must not error")
+	subs, err := b.Features.Subscriptions.ExpandTemplates(b)
```

Comment:
Any ability to test generation of auth subs?

Reply:
That's tested above in TestGenerateSubscriptions.
Just double checked to be sure.

Reply:
I misspoke, I meant to say subscribe to auth subs

Reply:
For live testing, right?
I think if you just set apiKey and apiSecret and then run with -tags mock_test_off it should JFW.

Do you think we need a mock test for it?
I just realised that we hit the live server WS either way, I think :'(

Reply:
Excuse my mess, but how about something like this?
```go
func TestSubscribe(t *testing.T) {
	t.Parallel()
	b := new(Bybit)
	require.NoError(t, testexch.Setup(b), "Test instance Setup must not error")
	b.Websocket.SetCanUseAuthenticatedEndpoints(true)
	subs, err := b.Features.Subscriptions.ExpandTemplates(b)
	require.NoError(t, err, "ExpandTemplates must not error")
	b.Features.Subscriptions = subscription.List{}
	mock := func(msg []byte, w *websocket.Conn) error {
		var req SubscriptionArgument
		require.NoError(t, json.Unmarshal(msg, &req), "Unmarshal should not error")
		require.Equal(t, req.Operation, "subscribe")
		msg, err := json.Marshal(SubscriptionResponse{
			Success:   true,
			RetMsg:    "OK",
			RequestID: req.RequestID,
			Operation: req.Operation,
		})
		require.NoError(t, err, "Marshal should not error")
		return w.WriteMessage(websocket.TextMessage, msg)
	}
	b = testexch.MockWsInstance[Bybit](t, testexch.CurryWsMockUpgrader(t, mock))
	b.Websocket.AuthConn = b.Websocket.Conn
	err = b.Subscribe(subs)
	require.NoError(t, err, "Subscribe must not error")
}
```

Reply:
Thank you 🙏 
Added failure case as well. Abused RetMsg very slightly in the positive case, but no big deal.

Fixed c5e16335